### ISSUE TITLE
Key F5 is assigned to 'run' and 'continue' simultaneously

### ISSUE THREAD

**Issue by laemmleint** [eclipse-theia/theia#14639]
### Bug Description:
The key F5 is assigned to 'run' and 'continue' by default. This leads to problems when it muss be decided which command should be applied.

### Steps to Reproduce:
Take this python script:
```
a = 1
b = 2
c = 3
print('end')
```

1. Make a breakpoint at line 3
2. Press and hold F5
3. Observe there are many debug sessions

![press_and_hold_f5](https://github.com/user-attachments/assets/ba396543-7f29-43da-a8b1-6b80629132f6)

### Additional Information

- Operating System: Windows 11
- Theia Version: 1.56.2

**Comment by msujew** [comment]
Looks like we're missing a `when` context here to prevent the command from running if we're already in a debug session:

https://github.com/eclipse-theia/theia/blob/4e3b183daab034ae86eaa610421d4d9791ffdaa5/packages/debug/src/browser/debug-frontend-application-contribution.ts#L1012-L1015

VS Code is doing this similarly, see [here](https://github.com/microsoft/vscode/blob/c269884f97e3ec4e62495ce64dec99d2d1f9db0b/src/vs/workbench/contrib/debug/browser/debugCommands.ts#L756-L760).

**Comment by laemmleint** [comment]
Thank you @msujew for the fast response.
I have a suggestion to change the Continue command:

```typescript
registry.registerCommand(DebugCommands.CONTINUE, {
    execute: () => {
        if (this.manager.state === DebugState.Stopped) {
            // eslint-disable-next-line no-unused-expressions
            this.manager.currentThread && this.manager.currentThread.continue();
        }
    },
    // When there is a debug session, F5 should always be captured by this command
    isEnabled: () => this.manager.state !== DebugState.Inactive
});
```

**Comment by msujew** [comment]
> It seems there is no 'registerCommandAndKeybindingRule' in the Theia code.

Well, they are two different services, so instead you would register commands and keybindings for those commands separately. 

> I have a suggestion to change the Continue command:

Note that the issue isn't so much that the `CONTINUE` command triggers, but rather that the `START` command triggers, even though a debugging session is already active.

Are you interested in contributing a pull request to this repo for the improvement?

**Comment by laemmleint** [comment]
Yes. I'm interested in contributing a pull request. I will open it in a couple of minutes...

**Comment by ankitsharma101** [comment]
Thanks for the detailed report and the VS Code reference @msujew!
I've opened a PR to fix this properly using when context keys as suggested.
Summary of the fix:

START keybinding: when: '!inDebugMode' → when: 'debugState == inactive'
CONTINUE keybinding: when: 'inDebugMode' → when: 'debugState == stopped'
Added a mutex lock in start() to guard against async race conditions during rapid key repeat
Updated DebugSessionManager.start() to immediately set context keys synchronously before async session setup, closing the timing window
Removed the isEnabled workaround from PR #14641 which is no longer needed

Tested by holding F5 for several seconds while paused at a breakpoint — only one session now remains active. PR: #17250
